spec/service_consumers/pact_helper.rb

```diff
@@ -106,6 +106,53 @@ def url_encode(str)
     end
   end
 
+  provider_state "there is a valid user session, with saved pages" do
+    set_up do
```

Comment:
I've had issues with these, its seems like they don't clear up behind themselves in the same way as before / after blocks.

So i've had state bleeding between my tests.
I've tried this and it works but I really don't like it. It feels messy and horrible